app/src/main/java/com/serwylo/lexica/view/BoardView.java

```diff
@@ -59,53 +63,72 @@ private boolean isCellHighlighted(int x, int y) {
 		return false;
 	}
 
+	private final Rect textBounds = new Rect();
+
 	@Override
 	public void onDraw(Canvas canvas) {
-		int width = getWidth();
-		int height = getHeight();
+		int width = getWidth() - paddingSize - 2;
+		int height = getHeight() - paddingSize - 2;
+
 
-		p.setARGB(255,255,255,255);
-		canvas.drawRect(0,0,width,height,p);
+		// Draw white box
+		p.setARGB(255, 255, 255, 255);
+		canvas.drawRect(paddingSize / 2, paddingSize, paddingSize / 2 + width, paddingSize + height, p);
 
-		if(board == null) {
+		if (board == null) {
 			return;
 		}
 
-		float boxSize = ((float) width) / board.getWidth();
+		float boxsize = ((float) height) / board.getWidth();
 
 		// Draw touched boxes
-		p.setARGB(255,255,255,0);
-		for (int x = 0; x < board.getWidth(); x ++) {
-			for (int y = 0; y < board.getWidth(); y ++) {
+		p.setARGB(255, 255, 255, 0);
+		for (int x = 0; x < board.getWidth(); x++) {
+			for (int y = 0; y < board.getWidth(); y++) {
 				if (!isCellHighlighted(x, y)) {
 					continue;
 				}
 
-				float left = boxSize * x;
-				float top = boxSize * y;
-				float right = boxSize * (x + 1);
-				float bottom = boxSize * (y + 1);
+				float left = paddingSize / 2 + boxsize * x;
```

Comment:
While correct, this line and the following lines could probably do with some extra parentheses, to clarify that it is actually `(paddingSize / 2) + (boxsize * x)`.